contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,570 @@
+        stakingModuleCache.treasuryFee = stakingModuleData.treasuryFee;
+        stakingModuleCache.targetShare = stakingModuleData.targetShare;
+        status = StakingModuleStatus(stakingModuleData.status);
+    }
+
+    /**
+     * @dev load not stopped modules list
+     * @notice used for reward distribution
+     * @return totalActiveKeys for not stopped modules
+     * @return modulesCache array of StakingModuleCache struct
+     */
+    function _loadNotStoppedStakingModulesCache() internal view returns (uint totalActiveKeys, StakingModuleCache[] memory modulesCache) {
+        uint256 modulesCount = _stakingModulesCount;
+        modulesCache = new StakingModuleCache[](modulesCount);
+        StakingModuleStatus status;
+        for (uint256 i; i < modulesCount; ++i) {
+            (status, modulesCache[i]) = _loadStakingModuleCache(i);
+
+            /// @dev account only keys from not stopped modules (i.e. active and paused)
+            if (status != StakingModuleStatus.Stopped) {
+                (modulesCache[i].activeKeysCount, modulesCache[i].availableKeysCount) = IStakingModule(modulesCache[i].stakingModuleAddress)
+                    .getKeysUsageData();
+                totalActiveKeys += modulesCache[i].activeKeysCount;
+            }
+        }
+    }
+
+    /**
+     * @dev load active modules list
+     * @notice used for deposits allocation
+     * @return totalActiveKeys for active modules
+     * @return modulesCache array of StakingModuleCache struct
+     */
+    function _loadActiveStakingModulesCache() internal view returns (uint totalActiveKeys, StakingModuleCache[] memory modulesCache) {
+        uint256 modulesCount = _stakingModulesCount;
+        modulesCache = new StakingModuleCache[](modulesCount);
+        StakingModuleStatus status;
+        for (uint256 i; i < modulesCount; ++i) {
+            (status, modulesCache[i]) = _loadStakingModuleCache(i);
+
+            /// @dev account only keys from active modules
+            if (status == StakingModuleStatus.Active) {
+                (modulesCache[i].activeKeysCount, modulesCache[i].availableKeysCount) = IStakingModule(modulesCache[i].stakingModuleAddress)
+                    .getKeysUsageData();
+                totalActiveKeys += modulesCache[i].activeKeysCount;
+            }
+        }
+    }
+
+    function _getKeysAllocation(
+        uint256 _keysToAllocate
+    ) internal view returns (uint256 allocated, uint256[] memory allocations, StakingModuleCache[] memory modulesCache) {
+        // calculate total used keys for operators
+        uint256 totalActiveKeys;
+        (totalActiveKeys, modulesCache) = _loadActiveStakingModulesCache();
+
+        uint256 modulesCount = modulesCache.length;
+        allocations = new uint256[](modulesCount);
+        /// @dev new estimated active keys count
+        totalActiveKeys += _keysToAllocate;
+        if (modulesCount > 0 && totalActiveKeys > 0) {
```

Comment:
totalActiveKeys??? Maybe return predicate from `_loadActiveStakingModulesCache ` -> atLeastOneModuleActive:bool

Reply:
totalAvailableKeys?

Reply:
just removed totalActiveKeys > 0, should be enough